datumaro/cli/commands/log.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2021 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import argparse
+
+from ..util.project import load_project
+
+
+def build_parser(parser_ctor=argparse.ArgumentParser):
+    parser = parser_ctor(description="Prints project history.")
+
+    parser.add_argument('-n', '--count', default=10, type=int,
+        help="Count of last commits to print (default: %(default)s)")
+    parser.add_argument('-p', '--project', dest='project_dir', default='.',
+        help="Directory of the project to operate on (default: current dir)")
+    parser.set_defaults(command=log_command)
+
+    return parser
+
+def log_command(args):
+    project = load_project(args.project_dir)
+
+    revisions = project.history(args.count)
+    if revisions:
+        print('\n'.join('%s %s' % line for line in revisions))
```

Comment:
It's unnecessary to build the combined string in memory. It would be more efficient (and IMO more readable) to just do:

```python
for line in revisions:
    print('%s %s' % line)
```

Reply:
Forgot about buffering -_-. Fixed.